interview-questions.md

```diff
@@ -4,49 +4,77 @@ Answer the following questions. First, without external resources. Challenge you
 
-  Your answer:
+  Your answer: Model View Controller
 
   Researched answer:
 
+Model
+The central component of the pattern. It is the application's dynamic data structure, independent of the user interface. It directly manages the data, logic and rules of the application.
+View
+Any representation of information such as a chart, diagram or table. Multiple views of the same information are possible.
+Controller
+Accepts input and converts it to commands for the model or view.
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: A gem is a special package of code that will assist your current code to do what it is assigned to do most efficiently. 
 
   Researched answer:
 
+In Ruby, a gem is a library that contains a specific piece of functionality as well as any files or assets related to that functionality. Gems can be found for all sorts of common functionality in an application -- things like handling money and currency, integrating with credit card processing, tools to make ruby coding even easier, and more. You could write these things yourself but one big benefit to using a gem instead is that it saves time.
 
 
 3. Why is it important to have validations in your application?
 
-  Your answer:
+  Your answer:to make sure the information your are inputting into your database can be properly acted upon with full information and no missing elements. 
 
   Researched answer:
 
+Validations are used to ensure that only valid data is saved into your database. For example, it may be important to your application to ensure that every user provides a valid email address and mailing address. Model-level validations are the best way to ensure that only valid data is saved into your database. They are database agnostic, cannot be bypassed by end users, and are convenient to test and maintain. Rails makes them easy to use, provides built-in helpers for common needs, and allows you to create your own validation methods as well.
 
 
 4. Describe a possible relationship between a model called Person and a model called Computer. Based on your description, which model would hold the foreign key?
 
-  Your answer:
+  Your answer: A person CAN HAVE MANY Computers // A Computer BELONGS TO a Person - The foreign key would belong to the Computer
 
   Researched answer:
+  The foreign key always goes on the belongs_to side. By convention the foreign key should always be the modelname_id and the data type should be integer. The has_many model is singular, the belongs_to model is plural. A Person (singular) has_many Computers (plural).
 
 
 
 5. What is object-relational mapping?
 
-  Your answer:
+  Your answer: I believe it relates to the belongs_to/has_many model.
 
   Researched answer:
+  
+  SELECT * FROM users WHERE email = 'test@test.com';
+Object-relational-mapping is the idea of being able to write queries like the one above, as well as much more complicated ones, using the object-oriented paradigm of your preferred programming language.
 
```

Comment:
Yes, for sure. ORM allows two different languages to interact with each other. In this case, Active Record (ORM) allows Rails to interact with Postgres.